networkx/algorithms/community/modularity_max.py

```diff
@@ -234,7 +215,94 @@ def greedy_modularity_communities(G, weight=None, resolution=1, n_communities=1)
             b[v] += b[u]
             b[u] = 0
 
-    return sorted(communities.values(), key=len, reverse=True)
+        yield communities.values()
+
+
+def greedy_modularity_communities(
+    G, weight=None, resolution=1, cutoff=1, best_n=None, n_communities=None
+):
+    """
+    Parameters
```

Comment:
I guess it is time to shift (or copy) the docstring info that is now in the private generator into this public function.

r"""Find communities in G using greedy modularity maximization.

    This function uses Clauset-Newman-Moore greedy modularity maximization [2]_.

    Greedy modularity maximization begins with each node in its own community
    and joins the pair of communities that most increases modularity until no
    such pair exists. Two keyword arguments adjust the stopping condition.
    `cutoff` is a lower limit on the number of communities so you can stop
    the process before reaching a maximum. `best_n` is an upper limit on the
    number of communities so you can make the process continue until at most
    n communities remain even if the maximum modularity occurs for more.
    To obtain exactly n communities, set both `cutoff` and `best_n` to n.

    This function maximizes the generalized modularity, where `resolution`
    is the resolution parameter, often expressed as $\gamma$.
    See :func:`~networkx.algorithms.community.quality.modularity`.
```

The references and See Also sections should also be copied to the public function.